src/hb-ot-layout.h

```diff
@@ -312,15 +312,17 @@ hb_ot_layout_collect_features (hb_face_t      *face,
 			       hb_tag_t        table_tag,
 			       const hb_tag_t *scripts,
 			       const hb_tag_t *languages,
-			       const hb_tag_t *features,
+			       const hb_set_t *features,
```

Comment:
We can't change the signature of an existing API method. Instead I would add a second version method that takes features as a set. Then that could internally convert features to an array and call this method, or you could update this method to convert the array to a set and call the new one.

Additionally I'd drop the retain_all_features bool and instead stick with the current behaviour that if the features set is a nullptr that means keep all.

Reply:
> if the features set is a nullptr that means keep all.

What if one wants to drop all features?

Reply:
Passing an empty set would drop all features.

Reply:
Right.